app/src/main/java/com/example/nvdovin/weatherapp/presentation/history/grid/dagger/GridHistoryFragmentModule.java

```diff
@@ -39,8 +40,8 @@ GridHistoryView provideGridHistoryView(GridHistoryFragment gridHistoryFragment,
 
     @Provides
     @GridHistoryFragmentScope
-    GridHistoryPresenter provideGridHistoryPresenter(CityService cityService, WeatherDataService weatherDataService, GridHistoryView gridHistoryView) {
-        return new GridHistoryPresenter(cityService, weatherDataService, gridHistoryView);
+    GridHistoryPresenter provideGridHistoryPresenter(DataMapper dataMapper, GridHistoryView gridHistoryView) {
```

Comment:
There is no need for this "@Provides
     @GridHistoryFragmentScope
     GridHistoryFragment provideGridHistoryFragment() {
         return gridHistoryFragment;
     }"

You can just use the gridHistoryFragment which you already have as a private variable (GridHistoryView)